src/foam/util/concurrent/AbstractAssembly.js

```diff
@@ -0,0 +1,87 @@
+/**
+ * @license
+ * Copyright 2019 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+foam.CLASS({
+  package: 'foam.util.concurrent',
+  name: 'AbstractAssembly',
+  //abstract: true,
+  // TODO?: Could provide a reset() method if wanted to make reusable to avoid GC.
+
+  /**
+   * Abstract implementation of Assembly interface.
+   * Provides concurrency support.
+   **/
+
+  implements: [
+    'foam.util.concurrent.Assembly'
+  ],
+
+  properties: [
+    {
+      name: 'isCompleted',
```

Comment:
Isn't synchronized will cause a bug.

Reply:
I guess I can't add "synchronized: true" to the property but I can make the setter/getter synchronized by overwriting it, and then the "complete" method wouldn't need to be synchronized